ios/listdevices.go

```diff
@@ -102,9 +103,47 @@ func (muxConn *UsbMuxConnection) ListDevices() (DeviceList, error) {
 	if err != nil {
 		return DeviceList{}, fmt.Errorf("Failed getting devicelist: %v", err)
 	}
+	muxResponse := MuxResponsefromBytes(response.Payload)
+	if muxResponse.Number == 4294967295 {
+		// get list of devices from Listen
+		// this is a workaround for the case where connected devices more than 34
+		return muxConn.ListDevicesFromListen()
+	}
 	return DeviceListfromBytes(response.Payload), nil
 }
 
+func (muxConn *UsbMuxConnection) ListDevicesFromListen() (DeviceList, error) {
+	attachedReceiver, err := muxConn.Listen()
+	if err != nil {
+		return DeviceList{}, fmt.Errorf("Failed to start listening: %v", err)
+	}
+	defer muxConn.Close()
+
+	deviceEntryChan := make(chan DeviceEntry)
+	go func() {
+		for {
+			msg, err := attachedReceiver()
+			if err != nil {
+				break
+			}
+			if msg.DeviceAttached() {
+				deviceEntryChan <- msg.DeviceEntry()
+			}
+		}
+	}()
+	var deviceList = DeviceList{
+		DeviceList: make([]DeviceEntry, 0, 40),
```

Comment:
why was 34 a limit? and why put 40 as the new limit?

Reply:
In most conditions, phone number is bellow 40 (from my experience).